### DISCUSSION TITLE
Pointer to "object pooling technique" for specific case

### DISCUSSION THREAD

**Discussion by IFB12** [ChebanovDD/MatchSweets#15] [General]
Hello

I dont understand this quote: "Note: The FallDownFillStrategy & SlideDownFillStrategy are given as an example. Consider to implement an `object pooling technique `for the ItemMoveData to reduce memory pressure". 

What is meant with "object pooling technique"? Can you give me a specific code example / pointer to it?

Cheers

**Comment by ChebanovDD** [comment]
Hi there,

Yes, if you want to avoid a memory allocation on every `GetSolveJobs` call, you have to consider implementing an object pooling for the `ItemMoveData` class, as I did for the `IUnityItem`'s.

You can learn more about object polling [here](https://learn.unity.com/tutorial/introduction-to-object-pooling#).
